pennylane/devices/default_qubit.py

```diff
@@ -349,6 +332,27 @@ def execute(
             # reset _rng to mimic serial behavior
             self._rng = np.random.default_rng(self._rng.integers(2**31 - 1))
 
+        if self.tracker.active:
+            self.tracker.update(batches=1)
+            self.tracker.record()
+            for c in circuits:
+                qpu_executions, shots = get_num_shots_and_executions(c)
+                if c.shots:
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        shots=shots,
+                        resources=c.specs["resources"],
+                    )
+                else:
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=results,
```

Comment:
Since `results` is for every circuit in the batch, should we update just the result that corresponds to the `i`th circuit?

Also, it looks like we need results for the finite shots case as well.

Reply:
updated, thanks!